File: Backend/core/models/keywordeffecttemplate.py
from django.db import models
from .trigger import Trigger
from .effect import Effect
from .condition import Condition
from .enums import TargetSpec, ListeningZone, TriggerScope, DynamicValueType, TargetingRule

class KeywordEffectTemplate(models.Model):
    name = models.CharField(max_length=50, unique=True)  # e.g. "LIFELINK", "PIERCE"
    trigger = models.ForeignKey(Trigger, on_delete=models.CASCADE)
    effect = models.ForeignKey(Effect, on_delete=models.CASCADE)
    value = models.CharField(
        max_length=50,
        choices=DynamicValueType.choices(),
        null=True,
        blank=True
    )
    targeting = models.CharField(max_length=30, choices=TargetSpec.choices, null=True, blank=True)
    targeting_rule = models.CharField(max_length=30, choices=TargetingRule.choices, null=True, blank=True)
    condition = models.ForeignKey(Condition, on_delete=models.SET_NULL, null=True, blank=True)
    condition_value = models.IntegerField(null=True, blank=True)

    zone = models.CharField(max_length=30, choices=ListeningZone.choices, default=ListeningZone.ANY)
    scope = models.CharField(max_length=30, choices=TriggerScope.choices, default=TriggerScope.SELF)

    linked_card = models.ForeignKey(
        "Card", on_delete=models.SET_NULL, null=True, blank=True, related_name="keyword_linked_bindings"
    )

    def __str__(self):
        return f"{self.name}"

File: Source/Eidolune/Core/KeywordEffectTemplate.cpp

#include "KeywordEffectTemplate.h"


    std::shared_ptr<Card> KeywordEffectTemplate::GetLinkedCard() const { return LinkedCard; }
    std::optional<int> KeywordEffectTemplate::GetConditionValue() const { return ConditionValue; }
    std::shared_ptr<Trigger> KeywordEffectTemplate::GetTrigger() const { return TriggerPtr; }

    std::shared_ptr<Effect> KeywordEffectTemplate::GetEffect() const { return EffectPtr; }
    std::optional<int> KeywordEffectTemplate::GetValue() const { return StaticValue; }
    std::string KeywordEffectTemplate::GetRawValue() const { return RawValue; }
    DynamicValueType KeywordEffectTemplate::GetValueType() const { return ValueType; }
    std::optional<TargetSpec> KeywordEffectTemplate::GetTargetSpec() const { return Targeting; }
    std::shared_ptr<Condition> KeywordEffectTemplate::GetCondition() const { return ConditionPtr; }

    TriggerScope KeywordEffectTemplate::GetScope() const { return Scope; }
    ListeningZone KeywordEffectTemplate::GetZone() const { return Zone; } 
